Incompatibility with TypeScript 4.8

## Describe the bug

As of TypeScript 4.8, `@restart/ui` is causing the following type error when `skipLibCheck` is `false`:

```
node_modules/@restart/ui/esm/usePopper.d.ts:2:69 - error TS2344: Type 'Options' does not satisfy the constraint 'Obj'.

2 export declare type Modifier<Name, Options> = Popper.Modifier<Name, Options>;
                                                                      ~~~~~~~

  node_modules/@restart/ui/esm/usePopper.d.ts:2:36
    2 export declare type Modifier<Name, Options> = Popper.Modifier<Name, Options>;
                                         ~~~~~~~

    This type parameter might need an `extends Popper.Obj` constraint.


Found 1 error in node_modules/@restart/ui/esm/usePopper.d.ts:2
```

## To Reproduce

Steps to reproduce the behavior:

1. Update TypeScript to 4.8.2 in a project using `@restart/ui` and `"skipLibCheck": false` in `tsconfig.json`
2. Run `tsc`

## Reproducible Example

https://stackblitz.com/edit/vitejs-vite-f1raze?file=package.json&terminal=build

## Expected behavior

I expect it to pass type checking.

## Screenshots

![image](https://user-images.githubusercontent.com/1288694/188195502-3a7fefdd-22b5-4862-9b50-f2f3d5d3420a.png)

## Environment (please complete the following information)

- Operating System: Windows, Linux, macOS
- Browser, Version: N/A
- react-overlays Version: N/A

## Additional context

The compiler's suggestion of adding `extends Popper.Obj` seems to resolve the issue but may be incompatible with the current usage as it appears [`undefined` may also be a valid value](https://github.com/react-restart/ui/blob/de7ce33ea3af51d77e589bf51f41ca071b3c0a6f/src/usePopper.ts#L58)? If that `undefined` can be replaced with something like `Record<string, never>` (empty object), then it may be fine.

--- a/src/usePopper.ts
+++ b/src/usePopper.ts
@@ -12,7 +12,10 @@ const disabledApplyStylesModifier = {
 };
 
 // until docjs supports type exports...
-export type Modifier<Name, Options> = Popper.Modifier<Name, Options>;
+export type Modifier<Name, Options extends Popper.Obj> = Popper.Modifier<
+  Name,
+  Options
+>;
 export type Options = Popper.Options;
 export type Instance = Popper.Instance;
 export type Placement = Popper.Placement;
@@ -55,7 +58,10 @@ export interface UsePopperState {
   state?: State;
 }
 
-const ariaDescribedByModifier: Modifier<'ariaDescribedBy', undefined> = {
+const ariaDescribedByModifier: Modifier<
+  'ariaDescribedBy',
+  Record<string, never>
+> = {
   name: 'ariaDescribedBy',
   enabled: true,
   phase: 'afterWrite',

